src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,42 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type == Bulldozer.class) {
+            List<Bulldozer> machines = new BulldozerProducer().get();
+            return new ArrayList<>(machines);
+        }
+        if (type == Excavator.class) {
+            List<Excavator> machines = new ExcavatorProducer().get();
+            return new ArrayList<>(machines);
+        }
+        if (type == Truck.class) {
+            List<Truck> machines = new TruckProducer().get();
+            return new ArrayList<>(machines);
+        }
```

Comment:
I had the same implemintation if... if.... if....
You should use if else here
And you don't need to create a new array list here, because your method get() has already created it